File: java-back-server/src/main/java/space/borisgk/taxi/api/model/entity/Message.java
package space.borisgk.taxi.api.model.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import space.borisgk.taxi.api.model.IEntity;
import space.borisgk.taxi.api.model.MessageStatus;
import space.borisgk.taxi.api.model.MessageType;

import javax.persistence.*;
import java.util.Date;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "message")
public class Message implements IEntity<Integer> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;
    @OneToOne
    @JoinColumn(name = "trip_id")
    private Trip trip;
    @Column(name = "data")
    private byte[] data;
    @Column(name = "date")
    private Date date;
    @Column(name = "status")
    private MessageStatus status;
    @Column(name = "type")
    private MessageType type;
}

File: kotlin-back-server/src/main/kotlin/space/borisgk/taxi/api/model/entity/Message.kt
package space.borisgk.taxi.api.model.entity

import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter

import javax.persistence.*
import java.util.Date

class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null
        set(id) {
            field = this.id
        }

    @OneToOne
    var user: User? = null
        set(user) {
            field = this.user
        }
    @OneToOne
    var trip: Trip? = null
        set(trip) {
            field = this.trip
        }
    var message: String? = null
        set(message) {
            field = this.message
        }
    var date: Date? = null
        set(date) {
            field = this.date
        }
}
